Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc

```diff
@@ -351,11 +359,17 @@ bool AlCaIsoTracksFilter::filter(edm::Event& iEvent, edm::EventSetup const& iSet
 				   << " Chg Isolation " << hmaxNearP
 				   << ":" << eIsolation;
 	  if (t_p>pTrackMin_ && eMipDR<eEcalMax_ && hmaxNearP<eIsolation) {
-	    accept = true;
-	    break;
+	    if (t_p > pTrackLow_ && t_p < pTrackHigh_) ntrin++;
+	    else                                       ntrout++;
 	  }
 	}
       }
+      accept = (ntrout > 0);
+      if (!accept && ntrin > 0) {
+	++nRange_;
+	if      (preScale_ <= 1)         accept = true;
+	else if (nRange_%preScale_ == 1) accept = true;
+      }
     }
   }
   // Step 4:  Return the acceptance flag
```

Comment:
Thanks @bsunanda for the clarifications.
Given that in this PR:

`pTrackMin_ == pTrackLow_
`
I follow your logic.